### ISSUE TITLE
testharness resends messages that we've seen

### ISSUE THREAD

**Issue by sefeng211** [web-platform-tests/wpt#16540]
Had this conversation with jgraham on IRC. It looks like https://searchfox.org/mozilla-central/rev/75294521381b331f821aad3d6b60636844080ee2/testing/web-platform/tests/resources/testharness.js#106-114 may possibly resend some messages that we've already seen. 

## Steps to reproduce
1) Modify [this line](https://searchfox.org/mozilla-central/rev/75294521381b331f821aad3d6b60636844080ee2/testing/web-platform/tests/FileAPI/url/sandboxed-iframe.html#30) to 
```
frame.onload = function() {
  fetch_tests_from_window(frame.contentWindow);
}
```
2) Run this test

**Comment by foolip** [comment]
@sefeng211 is the resulting in some tests being broken, or how did you stumble upon this?

cc @jgraham

**Comment by jgraham** [comment]
Yes, it results in tests being broken when you change the timing of `postMessage`. I made a patch but it broke the selftests, so I need to spend some more time on it.
